### DISCUSSION TITLE
Installation failed: Gateway Time-out

### DISCUSSION THREAD

**Discussion by hellofinch** [wavelog/wavelog#917] [Q&A] [answered]
I deploy the wavelog with docker. I followed the guide and tried to install the wavelog. I have tried many times and failed here. Something goes wrong.
The error is shown below.
![image](https://github.com/user-attachments/assets/0b223e69-eff3-4a22-841e-93a9c304f0f1)
I used Chrome DevTools to debug. There are some interesting things. 
![image](https://github.com/user-attachments/assets/6f3c93da-3834-42a5-ad00-ec880e2fbf8f)
As you can see, ajax.php keeps popping up. The installation is unfinished, but the web page has stopped.
Can I install the wavelog without `IP:PORT/install/`? Maybe I should wait?
THANKS!

**Comment by hellofinch** [comment]
BTW, I monitor the database and I saw new tables were created after the error pop up.
I think some timer is too short.

**Comment by HB9HIL** [comment]
Gateway Timed Out means that the webservers max_execution_time was reached. Did you adjust the max_execution_time regarding the recommendations?

If you use a reverse proxy in front of the webserver, you need to adjust the timeout at the proxy aswell.

**Reply by hellofinch** [reply]
Thanks for your reply. I adjust the max_execution_time to 3600. The error has changed.
![image](https://github.com/user-attachments/assets/cfcc6ea1-7f65-4123-a4ef-e8931d2692e3)
```
Wavelog Installer Debug Log
-------
Log Location: ../application/logs/installer_debug.log

DEBUG - 2024-09-10 03:00:52 --> Start writing config.php
ERROR - 2024-09-10 03:00:53 --> Failed to validate POST data
ERROR - 2024-09-10 03:00:53 --> File: Could not write file. Check Permissions.
DEBUG - 2024-09-10 03:04:09 --> Start writing config.php
DEBUG - 2024-09-10 03:04:10 --> File: config.php successfully written
DEBUG - 2024-09-10 03:04:10 --> Start writing database.php
DEBUG - 2024-09-10 03:04:11 --> File: database.php successfully written
DEBUG - 2024-09-10 03:04:11 --> Start creating database structure with assets/install.sql
DEBUG - 2024-09-10 03:04:32 --> Tables successfully created
DEBUG - 2024-09-10 03:04:32 --> Start migrating database to the newest version.
ERROR - 2024-09-10 03:09:32 --> Could not migrate database. Ajax crashed.
DEBUG - 2024-09-10 06:44:49 --> Start writing config.php
DEBUG - 2024-09-10 06:44:51 --> File: config.php successfully written
DEBUG - 2024-09-10 06:44:51 --> Start writing database.php
DEBUG - 2024-09-10 06:44:52 --> File: database.php successfully written
DEBUG - 2024-09-10 06:44:52 --> Start creating database structure with assets/install.sql
DEBUG - 2024-09-10 06:45:15 --> Tables successfully created
DEBUG - 2024-09-10 06:45:15 --> Start migrating database to the newest version.
ERROR - 2024-09-10 06:50:15 --> Could not migrate database. Ajax crashed.
DEBUG - 2024-09-10 07:00:27 --> Start writing config.php
DEBUG - 2024-09-10 07:00:28 --> File: config.php successfully written
DEBUG - 2024-09-10 07:00:28 --> Start writing database.php
DEBUG - 2024-09-10 07:00:29 --> File: database.php successfully written
DEBUG - 2024-09-10 07:00:29 --> Start creating database structure with assets/install.sql
DEBUG - 2024-09-10 07:00:54 --> Tables successfully created
DEBUG - 2024-09-10 07:00:54 --> Start migrating database to the newest version.
ERROR - 2024-09-10 07:07:29 --> Could not migrate database. Ajax crashed.
```
Where can I find the detailed info?
I put my install configuration here.
![image](https://github.com/user-attachments/assets/3e1103e7-1c7a-4685-a27c-f9431715c241)
![image](https://github.com/user-attachments/assets/1a0eb560-929a-46c5-9c41-844252486403)
![image](https://github.com/user-attachments/assets/4c397979-097e-4789-bfe7-4e9a2497fd3a)
![image](https://github.com/user-attachments/assets/ff133a87-572b-4fd5-8a3c-a1c7cdbe495e)
![image](https://github.com/user-attachments/assets/b5da619e-f5ef-443a-a710-457a6c4fe4b7)

**Reply by hellofinch** [reply]
If that is possible, can I install wavelog manually without PHP? Maybe some configurations in the webserver influences the installation process.

**Reply by HB9HIL** [reply]
It's pretty strange that this is happening in the docker setup. Do you use `latest` or `dev`?

**Comment by HB9HIL** [comment]
In fact it looks like a custom docker setup. You said you deploy Wavelog with docker. But I can see you are using nginx instead apache2, you are using a subfolder called "wavelog". At the end I can't say whats wrong here. The normal docker setup described in [here](https://github.com/wavelog/wavelog/wiki/Installation-via-Docker) is working properly 

Share your docker-compose.yml

**Reply by hellofinch** [reply]
Thanks for your reply. I tried to use docker compose mentioned [here](https://github.com/wavelog/wavelog/wiki/Installation-via-Docker) and failed. My Linux kernel is low, so apache2 doesn't work on my NAS. nginx is my only choice. So I don't use docker compose. I git clone the repo and follow [here](https://github.com/wavelog/Wavelog/wiki/Installation) to deploy.

**Reply by HB9HIL** [reply]
Did you follow the information regarding nginx and the special option? 

<img width="482" alt="image" src="https://github.com/user-attachments/assets/7a83d231-8c9e-403c-bc02-b087d300b87c">

**Reply by hellofinch** [reply]
Yes, I delete the `$`.
![image](https://github.com/user-attachments/assets/bee68050-9fe3-4c44-942f-34f469299eb6)
Here is my nginx configuration.

**Reply by HB9HIL** [reply]
@phl0 you are more used to nginx. Do you spot any issues here?

**Reply by phl0** [reply]
nginx config seems ok for me... Any errors in nginx log? Or in the network tab of dev console in browser?

**Reply by hellofinch** [reply]
Wow! I run install again and find a error in a lot of `ajax.php`.
![image](https://github.com/user-attachments/assets/f465fa48-1fa1-4212-96e1-493a8a1bded4)
![image](https://github.com/user-attachments/assets/a11e213d-1212-4c59-924f-e5452d9811bc)
```
root@fbraz3-lnmp1:/# tail /var/log/nginx/error.log  

root@fbraz3-lnmp1:/# cat /var/log/nginx/access.log | grep login
172.17.0.1 - - [12/Sep/2024:06:46:28 +0000] "GET /wavelog/index.php/user/login HTTP/1.1" 307 5 "http://100.105.183.10:8086/wavelog/install/run.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36"
172.17.0.1 - - [12/Sep/2024:06:46:28 +0000] "GET /wavelog/index.php/user/login HTTP/1.1" 307 5 "http://100.105.183.10:8086/wavelog/install/run.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36"
172.17.0.1 - - [12/Sep/2024:06:46:28 +0000] "GET /wavelog/index.php/user/login HTTP/1.1" 307 5 "http://100.105.183.10:8086/wavelog/install/run.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36"
172.17.0.1 - - [12/Sep/2024:06:46:28 +0000] "GET /wavelog/index.php/user/login HTTP/1.1" 307 5 "http://100.105.183.10:8086/wavelog/install/run.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36"
172.17.0.1 - - [12/Sep/2024:06:46:28 +0000] "GET /wavelog/index.php/user/login HTTP/1.1" 307 5 "http://100.105.183.10:8086/wavelog/install/run.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36"
172.17.0.1 - - [12/Sep/2024:06:46:28 +0000] "GET /wavelog/index.php/user/login HTTP/1.1" 307 5 "http://100.105.183.10:8086/wavelog/install/run.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36"
172.17.0.1 - - [12/Sep/2024:06:46:29 +0000] "GET /wavelog/index.php/user/login HTTP/1.1" 307 5 "http://100.105.183.10:8086/wavelog/install/run.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36"
172.17.0.1 - - [12/Sep/2024:06:46:29 +0000] "GET /wavelog/index.php/user/login HTTP/1.1" 307 5 "http://100.105.183.10:8086/wavelog/install/run.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36"
172.17.0.1 - - [12/Sep/2024:06:46:29 +0000] "GET /wavelog/index.php/user/login HTTP/1.1" 307 5 "http://100.105.183.10:8086/wavelog/install/run.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36"
172.17.0.1 - - [12/Sep/2024:06:46:29 +0000] "GET /wavelog/index.php/user/login HTTP/1.1" 307 5 "http://100.105.183.10:8086/wavelog/install/run.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36"
172.17.0.1 - - [12/Sep/2024:06:46:29 +0000] "GET /wavelog/index.php/user/login HTTP/1.1" 307 5 "http://100.105.183.10:8086/wavelog/install/run.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36"
172.17.0.1 - - [12/Sep/2024:06:46:29 +0000] "GET /wavelog/index.php/user/login HTTP/1.1" 307 5 "http://100.105.183.10:8086/wavelog/install/run.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36"
172.17.0.1 - - [12/Sep/2024:06:46:29 +0000] "GET /wavelog/index.php/user/login HTTP/1.1" 307 5 "http://100.105.183.10:8086/wavelog/install/run.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36"
172.17.0.1 - - [12/Sep/2024:06:46:29 +0000] "GET /wavelog/index.php/user/login HTTP/1.1" 307 5 "http://100.105.183.10:8086/wavelog/install/run.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36"
172.17.0.1 - - [12/Sep/2024:06:46:29 +0000] "GET /wavelog/index.php/user/login HTTP/1.1" 307 5 "http://100.105.183.10:8086/wavelog/install/run.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36"
172.17.0.1 - - [12/Sep/2024:06:46:29 +0000] "GET /wavelog/index.php/user/login HTTP/1.1" 307 5 "http://100.105.183.10:8086/wavelog/install/run.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36"
172.17.0.1 - - [12/Sep/2024:06:46:29 +0000] "GET /wavelog/index.php/user/login HTTP/1.1" 307 5 "http://100.105.183.10:8086/wavelog/install/run.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36"
172.17.0.1 - - [12/Sep/2024:06:46:29 +0000] "GET /wavelog/index.php/user/login HTTP/1.1" 307 5 "http://100.105.183.10:8086/wavelog/install/run.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36"
172.17.0.1 - - [12/Sep/2024:06:46:29 +0000] "GET /wavelog/index.php/user/login HTTP/1.1" 307 5 "http://100.105.183.10:8086/wavelog/install/run.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36"
172.17.0.1 - - [12/Sep/2024:06:46:29 +0000] "GET /wavelog/index.php/user/login HTTP/1.1" 307 5 "http://100.105.183.10:8086/wavelog/install/run.php" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36"
```
Here are some nginx log. I grep `login` for you.

**Reply by HB9HIL** [reply]
the amount of ajax.php is fully normal as it fetches some information for the UI every second during the installation.

But there is a "too many redirects" error.

**Reply by HB9HIL** [reply]
Can you check what kind of cookies are store currently in your browser? 

You can see them in the "App" tab of the dev tools

**Reply by hellofinch** [reply]
![image](https://github.com/user-attachments/assets/84b7144c-4cf6-4ab2-b772-22be7c103ed6)
![image](https://github.com/user-attachments/assets/f9578420-55c6-4665-af15-2cf19a65aa2c)

Here you are. THANKS!

**Reply by hellofinch** [reply]
BTW, I used to install cloudlog and there are also many "too many redirects" errors. I tried to ask for help but no one cared. At last, I failed to install cloudlog. :(

**Reply by HB9HIL** [reply]
For explaination:

a too many redirects - error happens when you have some sort of infinite loop. This can cause by a faulty webserver config or caused by code. 

The fact that you had the same issue in Cloudlog as now in Wavelog there is maybe a bug present which just occurs in special conditions or environment. We care about that and now our task is to find that. Or as mentioned it relies on your webserver configuration. I will try to reproduce your environment with nginx and will run some tests.

Meanwhile you could try to clear your browser from cache and cookies and try again. 

To reset an installation you have to perform the following tasks:

 - delete the file `.lock` in the `install/` folder if present
 - delete `application/config/config.php`
 - delete `application/config/database.php`
 - drop the database

**Reply by hellofinch** [reply]
Thanks for your reply. I reset the installation many times in the way you mentioned. I tried use a  brand new browser to install wavelog and failed too. I will try another browser to install wavelog. :)
I put my configuration files here to help you reproduce. I used [this](https://hub.docker.com/layers/fbraz3/lnmp/8.3/images/sha256-9604d4a41459041f165f228566df3ecfa3164072df0a96abfbf1bc09fdafa830?context=explore) docker image.
My `/etc/nginx/site-enabled/default` shows here:
```
server {
    listen 80 default_server;

    client_max_body_size 100M;

    root /app/public;
    index index.php index.html index.htm;

    location /pma/ {
        root /var/www/html;
        index index.php;

        location ~ \.php$ {
            fastcgi_buffer_size 128k;
            fastcgi_buffers 4 256k;
            fastcgi_busy_buffers_size 256k;

            fastcgi_pass unix:/run/php/php8.3-fpm.sock;
            fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
            fastcgi_param SCRIPT_NAME $fastcgi_script_name;
            include /etc/nginx/fastcgi_params;
            fastcgi_read_timeout 300;
        }
    }

    location /wavelog/ {
        root /var/www/html;
        index index.php;

        location ~ \.php {
            fastcgi_buffer_size 128k;
            fastcgi_buffers 4 256k;
            fastcgi_busy_buffers_size 256k;

            fastcgi_pass unix:/run/php/php8.3-fpm.sock;
            fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
            fastcgi_param SCRIPT_NAME $fastcgi_script_name;
            include /etc/nginx/fastcgi_params;
            fastcgi_read_timeout 3600;
        }
    }

    location / {
        try_files $uri $uri/ /index.php$is_args$args;
    }

    location ~ \.php$ {
        fastcgi_buffer_size 128k;
        fastcgi_buffers 4 256k;
        fastcgi_busy_buffers_size 256k;

        fastcgi_pass unix:/run/php/php8.3-fpm.sock;
        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
        fastcgi_param SCRIPT_NAME $fastcgi_script_name;
        include /etc/nginx/fastcgi_params;
        fastcgi_read_timeout 300;
    }

    location ~ /\.ht {
        deny all;
    }
}
```
And I adjust max_exection_time to 3600.
![image](https://github.com/user-attachments/assets/57316897-ceb2-49b0-a918-ea54de9c07d9)
I git clone wavelog to here shown in picture.
Any info you need please let me know. :)

**Reply by HB9HIL** [reply]
@hellofinch I was able to reproduce this issue! hang tight. I will investigate

**Reply by hellofinch** [reply]
I reinstalled again with a brand new browser and failed in the same place.
When error shows up, I try to visit `IP:PORT/wavelog/` and 302 `Too many redirects` shows up.

**Reply by hellofinch** [reply]
@HB9HIL Any idea how to solve the problem? :)

**Reply by HB9HIL** [reply]
@hellofinch 
Still try to figure out whats wrong during normal daily business. Please have some patience. This one is tricky

**Comment by HB9HIL** [comment]
@hellofinch 

This seems to be a more deep issue of the combination of codeigniter and nginx. I found several hints in the web how to handle nginx and a subfolder installation of codeigniter (like in your case). I have no clue whats the issue here as I'm usually using apache2. 

I asked @phl0 for help as he is more experienced with nginx. We will dig into this when we both have time. This is just to inform you that there is some progress ongoing. 

One thing which makes me concerned. Even if we find the issue. If your linux kernel on your NAS is too low for apache2. You will be have issues to run Wavelog anyway since the whole stuff needs some power, especially if you have a few thousand QSO's on it. We will try to find the nginx issue but you may should consider another host solution for your Wavelog Installation

**Reply by hellofinch** [reply]
Thanks for your reply. I will try another machine as you advice.
And I'm looking for any good info.
Have a nice day! 73.
:)

**Reply by HB9HIL** [reply]
Install it on another machine based on our recommendations with apache2!

**Reply by hellofinch** [reply]
I tried to install it on another machine and used docker-compose mentioned in WIKI.
I success!
I use my PC to run Docker. It works but is not convenient to use. If there is any good info for the bug we found, please let me know. 
Waiting for your good message~
Have a nice day. 73!

**Comment by HB9HIL** [comment, marked as answer]
@hellofinch Please see #1011 which tracks the progress of fixing this bug

**Comment by AndreasK79** [comment]
@hellofinch I see in your NGINX file that is says 

`location ~ \.php$ {`

Could you change that to

`location ~ \.php {`

restart NGINX and try again? The difference is that the trailing $ is removed.

**Comment by hellofinch** [comment]
Here is my solution.
I change my Nginx config. Here is my nginx configuration file.
```
server {
    listen 80 default_server;

    client_max_body_size 100M;

    root /app/public;
    index index.php index.html index.htm;

    location /pma/ {
        root /var/www/html;
        index index.php;

        location ~ \.php {
            fastcgi_buffer_size 128k;
            fastcgi_buffers 4 256k;
            fastcgi_busy_buffers_size 256k;

            fastcgi_pass unix:/run/php/php8.3-fpm.sock;
            fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
            fastcgi_param SCRIPT_NAME $fastcgi_script_name;
            include /etc/nginx/fastcgi_params;
            fastcgi_read_timeout 300;
        }
    }

    location /wavelog/ {
        root /var/www/html;
        # index index.php;
        try_files $uri $uri/ /wavelog/index.php?$args; # There is importance thing!

        location ~ \.php {
            fastcgi_buffer_size 128k;
            fastcgi_buffers 4 256k;
            fastcgi_busy_buffers_size 256k;

            fastcgi_pass unix:/run/php/php8.3-fpm.sock;
            fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
            fastcgi_param SCRIPT_NAME $fastcgi_script_name;
            include /etc/nginx/fastcgi_params;
            fastcgi_read_timeout 3600;
        }
    }

    location / {
        try_files $uri $uri/ /index.php$is_args$args;
    }

    location ~ \.php {
        fastcgi_buffer_size 128k;
        fastcgi_buffers 4 256k;
        fastcgi_busy_buffers_size 256k;

        fastcgi_pass unix:/run/php/php8.3-fpm.sock;
        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
        fastcgi_param SCRIPT_NAME $fastcgi_script_name;
        include /etc/nginx/fastcgi_params;
        fastcgi_read_timeout 300;
    }

    location ~ /\.ht {
        deny all;
    }
}
```
without `try_file`, the web browser will redirect many times.
After installation, `application/config/config.php` should modified.
change `config['index_page']='index.php' ` to `config['index_page']=''`
Then visit the web it will work.
BTW, although I success, DXCC update failed. I should update DXCC after login. Maybe it is my network fault. Just mentioned.
:)
@HB9HIL Maybe changing the answer? :)
